libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -980,6 +985,110 @@ void AP_DDS_Client::on_request(uxrSession* uxr_session, uxrObjectId object_id, u
         break;
     }
 #endif //AP_DDS_ARM_CHECK_SERVER_ENABLED
+#if AP_DDS_RALLY_SERVER_ENABLED
+    case services[to_underlying(ServiceIndex::GET_RALLY)].rep_id: {
+        ardupilot_msgs_srv_RallyGet_Request rally_get_request;
+        ardupilot_msgs_srv_RallyGet_Response rally_get_response;
+        const bool deserialize_success = ardupilot_msgs_srv_RallyGet_Request_deserialize_topic(ub, &rally_get_request);
+        if (deserialize_success == false) {
+            break;
+        }
+
+        const uxrObjectId replier_id = {
+            .id = services[to_underlying(ServiceIndex::GET_RALLY)].rep_id,
+            .type = UXR_REPLIER_ID
+        };
+
+        RallyLocation rally_location {};
+        const AP_Rally *rally = AP::rally();
+        if (rally->get_rally_point_with_index(rally_get_request.index, rally_location)) {
+            rally_get_response.success = true;
+            rally_get_response.size = rally->get_rally_total();
+            rally_get_response.rally.point.latitude = rally_location.lat * 1e-7;
```

Comment:
need to cast the 1e-7 to double or you will round to 1m, so do double(1.0e-7)
same in other places